File: leetcode-everyday/动态规划/基本动态规划-一维/70-爬楼梯/70-爬楼梯.js
/**
 * @param {number} n
 * @return {number}
 */
 var climbStairs = function(n) {
  let ans = 0;
  let prev = 0;
  let next = 1;
  for(let i = 1; i <= n ; ++i) {
      ans = prev + next;
      prev = next;
      next = ans;
  }
  return ans;
};

File: leetcode-everyday/动态规划/基本动态规划-一维/70-爬楼梯/70-爬楼梯.cpp
class Solution {
public:
    int climbStairs(int n) {
        int prev = 0;
        int next = 1;
        int ans = 0;
        for(int i = 1; i <= n; ++i) {
            ans = prev + next;
            prev = next;
            next = ans;
        }
        return ans;
    }
};